### PR TITLE
Fix settings handling

### PR DESCRIPTION
**Pull request by dcermak**

### CODE REVIEW

**Comment by okurz** [comment]
@AdamWill weren't there tests for this repo? https://github.com/os-autoinst/openQA-python-client/actions/runs/785550285 shows

```
actions/checkout@v1 and actions/setup-python@v1 are not allowed to be used in os-autoinst/openQA-python-client. Actions in this workflow must be: within a repository owned by os-autoinst.
```

any idea?

**Comment by dcermak** [comment]
Sounds like more restrictive permissions being applied here?

**Comment by AdamWill** [comment]
oh, I set permissions more restrictive because of a thing I read about where people are exploiting PRs to make github actions mine crypto, but apparently that won't work. guess I'll change it back.

File: tests/test_client.py [hunk 1]
```diff
@@ -167,6 +167,22 @@ def test_do_request_ok_yaml(self, fakesend, simple_config):
...
+        fakeresp = fakesend.return_value
+        fakeresp.status_code = 204
+        fakeresp.text = ""
+        client = oqc.OpenQA_Client()
+        request = requests.Request(
+            url=client.baseurl + "/api/v1/job_templates_scheduling/1", method="PUT"
+        )
+        ret = client.do_request(request)
+        # check we did not call .json() on the response (that would fail)
+        assert len(fakeresp.method_calls) == 0
```
Comment (okurz) [inline tests/test_client.py:182]: assert can take a second argument which I recommend to use, also see https://wiki.python.org/moin/UsingAssertionsEffectively , e.g.

```suggestion
        assert len(fakeresp.method_calls) == 0, 'invalid method call on response (or something)'
```

File: tests/test_client.py [hunk 2]
```diff
@@ -167,6 +167,22 @@ def test_do_request_ok_yaml(self, fakesend, simple_config):
...
+        # check that we returned the request object
+        assert ret == fakesend.return_value
```
Comment (okurz) [inline tests/test_client.py:184]: same here then

**Comment by okurz** [comment]
@AdamWill yeah we actually had that case of crypto miner exploitation PRs on os-autoinst

File: tests/test_client.py [hunk 3]
```diff
@@ -167,6 +167,22 @@ def test_do_request_ok_yaml(self, fakesend, simple_config):
...
+        fakeresp.status_code = 204
+        fakeresp.text = ""
+        client = oqc.OpenQA_Client()
+        request = requests.Request(
+            url=client.baseurl + "/api/v1/job_templates_scheduling/1", method="PUT"
+        )
+        ret = client.do_request(request)
+        # check we did not call .json() on the response (that would fail)
+        assert len(fakeresp.method_calls) == 0, "no methods must be called on response"
+        # check that we returned the request object
```
Comment (okurz) [inline tests/test_client.py:183]: I meant that you actually don't need the comments anymore if the assert call speaks for itself :)

**Review by okurz — CHANGES_REQUESTED** [review]

**Review by okurz — APPROVED** [review]

**Comment by okurz** [comment]
@AdamWill can you do a new release on PyPi of https://github.com/os-autoinst/openQA-python-client/releases/tag/4.1.2 ?

**Comment by AdamWill** [comment]
Er, well it's a bit harder now you've tagged it :( You also didn't update `CHANGELOG.md`. There's a `release.sh` script in the repo which is supposed to be used to do releases. Can you please just ping me next time?

**Comment by AdamWill** [comment]
you didn't update `setup.py` or `__init__.py` either...all the things the script does...

**Comment by AdamWill** [comment]
okay, I wiped the 4.1.2 tag and started over, release should be done correctly now.

**Comment by okurz** [comment]
sorry

**Pull request merged**